roles/reportportal/tasks/main.yaml

```
@@ -0,0 +1,81 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check if mandatory variables are defined and not empty
+  ansible.builtin.assert:
+    that:
+      - cifmw_reportportal_datarouter_url is defined and cifmw_reportportal_datarouter_url | length > 0
+      - cifmw_reportportal_datarouter_username is defined and cifmw_reportportal_datarouter_username | length > 0
+      - cifmw_reportportal_datarouter_password is defined and cifmw_reportportal_datarouter_password | length > 0
+      - cifmw_reportportal_project is defined and cifmw_reportportal_project | length > 0
+      - cifmw_reportportal_droute_client_url is defined and cifmw_reportportal_droute_client_url | length > 0
+
+- name: Url for downloading data router client
+  ansible.builtin.set_fact:
+    droute_url: >-
+      {{
+        (cifmw_reportportal_droute_client_url,
+         cifmw_reportportal_droute_version,
+         cifmw_reportportal_droute_binary) |
+        path_join
+      }}
+
+- name: Make sure the droute directory exists
+  ansible.builtin.file:
+    path: "{{ cifmw_reportportal_droute_dir }}"
+    state: directory
+    mode: "0755"
+
+- name: Get the Data router client
```

We try avoid setting facts when we can. I would suggest this:
```suggestion
- name: Get the Data router client
  vars:
    _droute_url: >-
      {{
        (cifmw_reportportal_droute_client_url,
        cifmw_reportportal_droute_version,
        cifmw_reportportal_droute_binary) |
        path_join
      }}
```

This allows you to remove the `- name: Url for downloading data router client` task and the `_` prefix indicates `_droute_url` is internal and not for user consumption (and not documented in the readme etc.)

---

Done.